### PR TITLE
Screenshot on each steps in UI tests

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Add an option in UI tests to screenshot each step, also add a longer timeout to wait for alias input in hummingbird theme
| Type?             | bug fix
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI green and UI tests green
| UI Tests          | https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/10615986213
| Fixed issue or discussion?     | ~
| Related PRs       | ~
| Sponsor company   | ~

### Explanation of the changes in hummingbird tests

While testing the classic edition it turned out the humming bird tests were failing quite often, but it turned out to be a "random" failure, sometimes with the exact same code the test could be green (although rare)

Failing test: https://github.com/PrestaShopCorp/ga.tests.ui.edition/actions/runs/10324267971
Successful test: https://github.com/PrestaShopCorp/ga.tests.ui.edition/actions/runs/10323729807

So the instability relied on the fact the alias input was not visible early in the process, although present on that page and the filling of this field was skipped sometimes I don't know why this bug doesn't appear while testing the open source version because the classic build used in this test has no modules therefore no differences with the core and yet it's more sensitive on this test

Edit: Also here is a campaign that was run on the core and had the same bug https://github.com/jolelievre/ga.tests.ui.pr/actions/runs/10388401566/job/28764206985 So the instability is not only with the classic build but also on the core

### CODE REVIEW

File: .github/actions/ui-test/action.yml [hunk 1]
```diff
@@ -30,6 +30,10 @@ inputs:
     required: false
     description: Database Server (mysql/mariadb)
     default: 'mysql'
+  SCREENSHOT_EACH_STEP:
```
Comment (Progi1984) [inline .github/actions/ui-test/action.yml]:
```suggestion
  TAKE_SCREENSHOT_AFTER_EACH_STEP:
```

File: tests/UI/package.json [hunk 2]
```diff
@@ -110,6 +110,7 @@
     "chai": "^4.5.0",
     "chai-string": "^1.5.0",
     "dotenv": "^16.4.5",
+    "lodash": "^4.17.21",
```
Comment (Progi1984) [inline tests/UI/package.json]: Remove this dependency, please

File: tests/UI/utils/setup.ts [hunk 3]
```diff
@@ -47,25 +48,37 @@ after(async function () {
...
-    }
-
+    await takeScreenShotAfterStep(this.browser, `${global.SCREENSHOT.FOLDER}/fail_test_${screenshotNumber}.png`);
     screenshotNumber += 1;
   }
+  if (global.SCREENSHOT.EACH_STEP) {
+    const testPath = this.currentTest?.file;
+    // eslint-disable-next-line no-unsafe-optional-chaining
+    const folderPath = testPath?.slice(testPath?.indexOf('tests/UI') + 8).slice(0, -3);
+    const screenshotPath = `${global.SCREENSHOT.FOLDER}${folderPath}/${_.kebabCase(this.currentTest?.title)}.png`;
```
Comment (Progi1984) [inline tests/UI/utils/setup.ts]: Add a function `kebabCase` to the `ui-testing-library` And update the library in the Core

**Review by Progi1984 — CHANGES_REQUESTED** [review]
Some feedbacks

File: .github/actions/ui-test/action.yml [hunk 4]
```diff
@@ -42,7 +46,10 @@ runs:
     - name: Use .env.ci
       working-directory: ${{ inputs.TESTS_DIR }}
       shell: bash
-      run: cp .env.ci .env
+      run: |
+        cp .env.ci .env
+        sed -i -e "s/TAKE_SCREENSHOT_AFTER_EACH_STEP.*/TAKE_SCREENSHOT_AFTER_EACH_STEP=${{ inputs.TAKE_SCREENSHOT_AFTER_EACH_STEP == 'false' && 'false' || 'true' }}/g" .env
+        cat .env
```
Comment (Progi1984) [inline .github/actions/ui-test/action.yml]:
```suggestion
```

**Review by Progi1984 — COMMENTED** [review]
A small feedback and it's ok

**Comment by jolelievre** [comment]
Thanks for the review @Progi1984 
Everything should be fixed now, and I see you already merged the PR on UI tests pr repo, only this one missing for the feature to work 😉

**Comment by Progi1984** [comment]
@jolelievre I fixed `ga.tests.ui.pr`. Sorry, i misread.

**Comment by jolelievre** [comment]
> @jolelievre I fixed `ga.tests.ui.pr`. Sorry, i misread.

Ah you're right, I forgot to remove the last cat that was meant for debug initially, no PR though 😅

**Review by Progi1984 — APPROVED** [review]

**Review by matthieu-rolland — APPROVED** [review]

**Pull request merged**